docs/standard/microservices-architecture/microservice-ddd-cqrs-patterns/implement-value-objects.md

```diff
@@ -120,25 +122,170 @@ public class Address : ValueObject
+
+With EF Core 2.0 there are new and better ways to persist value objects.
+
+## Persist value objects as owned entity types in EF Core 2.0
+
+Even when there are still some gaps between the canonical value object pattern in DDD and the owned entity type in EF Core (see limitations at the end of this section), it is currently, with EF Core 2.0, the best way to persist value objects.
+
+The onwed entity type feature was added to EF Core since version 2.0.
+
+An owned entity type allows you to map types that do not have their own identity explicitely defined in the domain model and are used as properties (such as a value object) within any of your entities. An owned entity type shares the same CLR type with another entity type. The entity containing the defining navigation is the owner entity. When querying the owner, the owned types will be included by default.
+
+Just by looking at the domain model, an owned type looks like it doesn’t have any identity but, under the covers, owned types do have identity, but the owner navigation property is part of this identity.
+
+The identity of instances of own types is not completely their own. It consists of three components: they identity of the owner, the navigation property pointing to them, and in the case of collections of owned types (not supported yet in EF Core 2.0), an independent component.
+
+For example, in the Ordering domain model at eShopOnContainers, as part of the Order entity, the Address value object is implemented as an owned entity type within the owner entity which is the Order entity. Address is a type with no identity property defined in the domain model. It is used as a property of the Order type to specify the shipping address for a particular order.
+
+By convention, a shadow primary key will be created for the owned type and it will be mapped to the same table as the owner by using table splitting. This allows to use owned types similarly to how complex types are used in EF6 in the traditional .NET Framework.
+
+It is important to note that owned types are never discovered by convention in EF Core, so you have to declare them explicetely.
+
+In eShopOnContainers, at the OrderingContext.cs, within the OnModelCreating() method, there are multiple infrastructure confirguration being applied. One of them is related to the Order entity.
+
+```csharp
+// Part of the OrderingContext.cs class at the Ordering.Infrastructure project
+// 
+protected override void OnModelCreating(ModelBuilder modelBuilder)
+{
+    modelBuilder.ApplyConfiguration(new ClientRequestEntityTypeConfiguration());
+    modelBuilder.ApplyConfiguration(new PaymentMethodEntityTypeConfiguration());
+    modelBuilder.ApplyConfiguration(new OrderEntityTypeConfiguration());
+    modelBuilder.ApplyConfiguration(new OrderItemEntityTypeConfiguration());
+    //...Additional type configurations
+}
+```
+
+In the following code is where the persistence infrastructure is defined for the Order entity.
```

Comment:
In the following code, the persistence infrastructure is defined for the Order entity:

Reply:
👍